HPXMLtoOpenStudio/resources/hvac_sizing.rb

```diff
@@ -1886,87 +1886,69 @@ def self.apply_hvac_ground_loop(hvac_sizing_values, weather, hvac_cooling)
-                      'l-config' => [3, 4, 5, 6],
-                      'rectangle' => [2, 4, 6, 8],
-                      'u-config' => [5, 7, 9],
-                      'l2-config' => [8],
-                      'open-rectangle' => [8] }
-    valid_num_bores = valid_configs[bore_config]
-    max_valid_configs = { 'line' => 10, 'l-config' => 6 }
-    unless valid_num_bores.include? num_bore_holes
-      # Any configuration with a max_valid_configs value can accept any number of bores up to the maximum
-      if max_valid_configs.keys.include? bore_config
-        max_num_bore_holes = max_valid_configs[bore_config]
-        num_bore_holes = max_num_bore_holes
-      else
-        # Search for first valid bore field
-        new_bore_config = nil
-        valid_configs.keys.each do |bore_config|
-          next unless valid_configs[bore_config].include? num_bore_holes
+    geothermal_loop = hvac_cooling.geothermal_loop
+    bore_spacing = geothermal_loop.bore_spacing
+    bore_diameter = geothermal_loop.bore_diameter
+    grout_conductivity = geothermal_loop.grout_conductivity
+    pipe_r_value = gshp_hx_pipe_rvalue(hvac_cooling)
+    nom_length_heat, nom_length_cool = gshp_hxbore_ft_per_ton(weather, hvac_cooling_ap, bore_spacing, bore_diameter, grout_conductivity, pipe_r_value)
 
-          new_bore_config = bore_config
-          break
-        end
-        if not new_bore_config.nil?
-          bore_config = new_bore_config
-        else
-          fail 'Could not construct a valid GSHP bore field configuration.'
+    loop_flow = geothermal_loop.loop_flow
+    if loop_flow.nil?
+      loop_flow = [1.0, UnitConversions.convert([hvac_sizing_values.Heat_Capacity, hvac_sizing_values.Cool_Capacity].max, 'Btu/hr', 'ton')].max.floor * 3.0
+    end
+
+    num_bore_holes = geothermal_loop.num_bore_holes
+    if num_bore_holes.nil?
+      num_bore_holes = [1, (UnitConversions.convert(hvac_sizing_values.Cool_Capacity, 'Btu/hr', 'ton') + 0.5).floor].max
+    end
+
+    min_bore_depth = UnitConversions.convert(24, 'm', 'ft')
+    max_bore_depth = UnitConversions.convert(384, 'm', 'ft')
+
+    bore_depth = geothermal_loop.bore_length
+    if bore_depth.nil?
+      bore_length_heat = nom_length_heat * hvac_sizing_values.Heat_Capacity / UnitConversions.convert(1.0, 'ton', 'Btu/hr')
+      bore_length_cool = nom_length_cool * hvac_sizing_values.Cool_Capacity / UnitConversions.convert(1.0, 'ton', 'Btu/hr')
+      bore_length = [bore_length_heat, bore_length_cool].max
+      bore_depth = (bore_length / num_bore_holes).floor # ft
+
+      for _i in 0..4
+        if (bore_depth < min_bore_depth) && (num_bore_holes > 1)
+          num_bore_holes -= 1
+          bore_depth = (bore_length / num_bore_holes).floor
+        elsif bore_depth > max_bore_depth
+          num_bore_holes += 1
+          bore_depth = (bore_length / num_bore_holes).floor
         end
       end
+
+      bore_depth = (bore_length / num_bore_holes).floor + 5 # FIXME: why add the 5?
```

Comment:
Pretty sure I know why. Soil temperature close to the surface is pretty close to the air temperature, then it gets more constant the deeper you go. So the active length, which is what we really care about, starts about 5 ft below the surface. That's what the 5 is for.

Reply:
"The g-functions produced by Eskilson use an assumed value of D/H that is not
clearly spelled out by Eskilson, though the numerical examples in the thesis all use
a depth D of 4e5 m. It is worth noting that, for grouted boreholes, the depth D can
represent the depth at which the horizontal connecting piping is buried and so the depth
at which the GHE “starts.” Or, for groundwater-filled boreholes, it can represent the
depth of the water table. Above that depth, the borehole is filled with air and the borehole
resistance is so high that the GHE is effectively inactive." From Chapter 2 of https://www.sciencedirect.com/book/9780081003114/advances-in-ground-source-heat-pump-systems.